app/terminal/terminal.tsx

```diff
@@ -97,7 +97,7 @@ export default class TerminalComponent extends React.Component<TerminalProps, St
     window.removeEventListener("keydown", this.windowKeyDownListener);
   }
 
-  componentDidUpdate(_prevProps: TerminalProps, prevState: State, snapshot: Snapshot | null) {
+  componentDidUpdate(_prevProps: TerminalProps, prevState: State, snapshot: Snapshot) {
```

Comment:
I think maybe this needs to be `snapshot?: Snapshot` based on the type signature here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/522a694de4e20d1ab69598affc0a97d2ee8d2556/types/react/v16/index.d.ts#L681

I remember hitting an issue that snapshot was either null or undefined which is why there's the `if (snapshot && ...)` check below

Reply:
Ahh yeah good catch.